transport-native-epoll/src/main/java/io/netty/channel/epoll/LinuxSocket.java

```diff
@@ -71,6 +73,41 @@ void setNetworkInterface(NetworkInterface netInterface) throws IOException {
         setInterface(intValue(), ipv6, nativeAddress.address(), nativeAddress.scopeId(), interfaceIndex(netInterface));
     }
 
+    InetAddress getInterface() throws IOException {
+        NetworkInterface inf = getNetworkInterface();
+        if (inf != null) {
+            Enumeration<InetAddress> addresses = SocketUtils.addressesFromNetworkInterface(inf);
+            if (addresses.hasMoreElements()) {
+                return addresses.nextElement();
+            }
+        }
+        return null;
+    }
+
+    NetworkInterface getNetworkInterface() throws IOException {
+        int ret = getInterface(intValue(), ipv6);
+        if (ipv6) {
+            return PlatformDependent.javaVersion() >= 7 ? NetworkInterface.getByIndex(ret) : null;
```

Comment:
I'd expect this to fail on Java 6, because as I remember dependent methods are loaded when the method is run, not based on whether the actual line using that method is run.

Reply:
TIL... I thought this is not the case as we also use the same patter in NioDatagramChannel to not support. Are you sure @ejone86 ?

Reply:
If you have read Java Puzzlers, it shows how this can fail. It's due to how the byte code does type checking on local variables.  It makes the claim that the *only* safe way to do this is via reflection.

I don't normally comment on the version checking, because it seems to work.  But from a correctness standpoint, I think it will fail.  I think @ejona86 is technically wrong in this case, because it isn't methods that are loaded, but classes.  You are still on shaky ground though.

Reply:
I see... As we use the same pattern else where in the code are you both fine to just merge @carl-mastrangelo @ejona86  ?

Reply:
I just tested it with a toy problem, and this pattern seemed to work in Java 6. So maybe it is a class vs method thing or a "get lucky" thing. (For reference, I didn't mean "loaded" like a class is loaded; I meant the method was loaded into whatever data structures Java uses for this calling method, so it was looked up/retrieved.)

I do remember that the JLS said that such loading was undefined so "all bets are off."